src/hffs/fs.py

```diff
@@ -155,33 +188,36 @@ def _open(
         self,
         path: str,
         mode: str = "rb",
+        revision: Optional[str] = None,
         **kwargs,
     ):
         if mode == "ab":
             raise NotImplementedError("Appending to remote files is not yet supported.")
         path = self._strip_protocol(path)
-        return HfFile(self, path, mode=mode, **kwargs)
+        return HfFile(self, path, mode=mode, revision=revision, **kwargs)
 
-    def _rm(self, path, **kwargs):
-        repo_type, repo_id, path_in_repo = self._resolve_repo_id(path)
+    def _rm(self, path, revision: Optional[str] = None, **kwargs):
         path = self._strip_protocol(path)
-        operations = [huggingface_hub.CommitOperationDelete(path_in_repo=path_in_repo)]
+        resolved_path = self.resolve_path(path, revision=revision)
+        operations = [huggingface_hub.CommitOperationDelete(path_in_repo=resolved_path.path_in_repo)]
         commit_message = f"Delete {path}"
         self._api.create_commit(
-            repo_id=repo_id,
-            repo_type=repo_type,
+            repo_id=resolved_path.repo_id,
+            repo_type=resolved_path.repo_type,
             token=self.token,
             operations=operations,
-            revision=self.revision,
+            revision=revision,
```

Comment:
If revision is None, default back to the resolved path one just in case?

```suggestion
            revision=revision or resolved_path.revision,
```